proxy/context.go

```diff
@@ -35,14 +36,19 @@ type context struct {
 	outgoingHost         string
 	debugFilterPanics    []interface{}
 	outgoingDebugRequest *http.Request
-	loopCounter          int
+	executionCounter     int
 	startServe           time.Time
 	metrics              *filterMetrics
 	tracer               opentracing.Tracer
 	proxySpan            opentracing.Span
 	parentSpan           opentracing.Span
+	proxy                *Proxy
+	routeLookup          *routing.RouteLookup
+}
 
-	routeLookup *routing.RouteLookup
+type teeTie struct {
```

Comment:
Can we please move the teeTie type to its own file? There's quite some concept bundled already in context.go.

Reply:
Moved.

Reply:
This is done.